test/es6/proxybugs.js

```diff
@@ -378,11 +378,25 @@ var tests = [
                 }
             };
 
-            var obj = Proxy.revocable(() => {}, handler);
+            var obj = Proxy.revocable(function() {}, handler);
             new obj.proxy();
             assert.isTrue(trapCalled);
         }
     },
+    {
+        name: "Assertion validation : proxy of arrow function should not be constructable",
+        body() {
+            let obj = new Proxy(()=>{}, {construct: ()=>({})});

```

Comment:
We should test a few more cases like non-constructible builtins (Math.abs, eval, parseInt, etc), 

class members:
```js
class a { 
  static foo() { } 
  bar() { } 
};
new a.foo();
new (new a()).bar();
```
object literal methods:
```js
var obj = { foo() { } };
new obj.foo();
```